extensions/kafka-streams/deployment/src/main/java/io/quarkus/kafka/streams/deployment/KafkaStreamsProcessor.java

```diff
@@ -56,7 +56,19 @@ void build(RecorderContext recorder,
         reflectiveClasses.produce(new ReflectiveClassBuildItem(true, false, false, ByteArraySerde.class));
         reflectiveClasses.produce(new ReflectiveClassBuildItem(true, false, false, FailOnInvalidTimestamp.class));
 
-        nativeLibs.produce(new SubstrateResourceBuildItem(Environment.getJniLibraryFileName("rocksdb")));
+        // for RocksDB, either add linux64 native lib when targeting containers
+        String containerRuntime = System.getProperty("native-image.container-runtime");
```

Comment:
That's not the only flag. You need to check for `-Dnative-image.docker-build` too

Reply:
@gsmet @dmlloyd @stuartwdouglas Maybe we need a flag as we have for modes. WDYT?

Reply:
Yes, I didn't feel happy about this specific solution. I'll add the check for the other property, too, as you suggested (btw. why do they both exist, there seems to be some overlap?

Would be great to have this information in a more accessible way. Perhaps it could be exposed via `RecorderContext`?

Reply:
This doesn't make sense for `RecorderContext`, which relates to bytecode generation.

I think maybe we should have a `NativeImageSetupBuildItem` which contains information about the native image configuration.  Then you could have a separate build step which consumes that and produces the substrate resource build item.  If the step produces nothing else, it won't even run if there's no native image being produced.

Reply:
I've logged https://github.com/quarkusio/quarkus/issues/2827 for adding an PI which makes that information accessible to extensions.

Reply:
Reworked this commit to also check `-Dnative-image.docker-build`.